src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java

```diff
@@ -53,21 +61,30 @@ public static void generateXdocContent() throws Exception {
         for (Path path : templatesFilePaths) {
             final String pathToFile = path.toString();
             final File inputFile = new File(pathToFile);
-            final File outputFile = new File(pathToFile.replace(".template", ""));
-
+            final File tempFile = File.createTempFile(pathToFile.replace(".template", ""), "");
             final XdocsTemplateSinkFactory sinkFactory = (XdocsTemplateSinkFactory)
                     plexus.lookup(SinkFactory.ROLE, XDOCS_TEMPLATE_HINT);
-            final Sink sink = sinkFactory.createSink(outputFile.getParentFile(),
-                    outputFile.getName(), String.valueOf(StandardCharsets.UTF_8));
+            final Sink sink = sinkFactory.createSink(tempFile.getParentFile(),
+                    tempFile.getName(), String.valueOf(StandardCharsets.UTF_8));
             final XdocsTemplateParser parser = (XdocsTemplateParser)
                     plexus.lookup(Parser.ROLE, XDOCS_TEMPLATE_HINT);
             try (Reader reader = ReaderFactory.newReader(inputFile,
                     String.valueOf(StandardCharsets.UTF_8))) {
                 parser.parse(reader, sink);
             }
             finally {
+
                 sink.close();
             }
+
+            final File outputFile = new File(pathToFile.replace(".template", ""));
+
+            final Object exp = StandardCopyOption.REPLACE_EXISTING;
+
+            Files.copy(tempFile.toPath(), outputFile.toPath(), (CopyOption) exp);
```

Comment:
Please do `Files.copy(tempFile.toPath(), outputFile.toPath(), StandardCopyOption.REPLACE_EXISTING);`, assignment to `Object` then casting to `CopyOption` seems strange.

Reply:
@nrmancuso Actually it is exceeding the word limit that's why i split it, and causing me  an error.

Reply:
`final StandardCopyOption exp = StandardCopyOption.REPLACE_EXISTING`

Reply:
Done @romani

Reply:
Resolved

Reply:
@romani Can you please check this comment https://github.com/checkstyle/checkstyle/issues/13894#issuecomment-1763632489. Sorry for adding in the PR